src/util/util.js

```diff
@@ -175,10 +176,11 @@ function encodeURIComponentRFC3986(str) {
   );
 }
 
-function leftpad(str) {
-  if (str.length < 8) {
+function leftpad(str, len) {
+  str = String(str);
+  if (str.length < len) {
     return (
-      range(0, 8 - str.length, false)
+      range(0, len - str.length, false)
```

Comment:
First of all thank you for this PR, and for the contribution.
WDYT to put the second parameter as optional ? with value to 8, as before ?
Also I think we shouldn't reaffect `str`
I would like to propose this, if you are ok:

`function leftpad(input, len = 8) {
const str = String(input);`

Then the previous call to `leftpad` stay the same (and if anyone was using this outside the library, it will not break the code)

Reply:
I thought that those utils are an implementation detail of vast-client and are not exposed. Your proposal makes sense and I'll adjust the pull as suggested.

Reply:
Adjusted it.